### ISSUE TITLE
Unhandled exception. System.IO.IOException: Read-only file system

### ISSUE THREAD

**Issue by Phylloxera** [sylabs/singularity#218]
**Type of issue**
test problem 

**Description of issue**
I've got a dotnet dependent app that I'm trying to containerize. I've run the app both as user and root apart from singularity. I've figured out how to make it work with a sandbox intermediate. But can't figure out how to design a recipe that makes a working container without that intermediate sandbox step.
My sandbox recipe:
BootStrap: docker
From: ubuntu:18.04
%files
/my/test/data.zip /home
%environment
export LC_ALL=C
%post
```
apt -y -qq update && apt -y -qq install unzip wget
wget https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
dpkg -i packages-microsoft-prod.deb; rm packages-microsoft-prod.deb
apt update; apt upgrade; apt -y -qq install aspnetcore-runtime-5.0
cd /home; wget http://appIwant.zip; unzip appIwant.zip; unzip data.zip; mv 'app I want' appIwant; rm *.zip
chmod 111 binary; chmod 775 <appIwant folder and subfolders>; mkdir /usr/msfind; mv appIwant /usr/msfind
mv /home/data /run
```

runs fine:
`sudo singularity build --sandbox sandbox/intermediate recipe.def`
gives "Unhandled exception. System.IO.IOException: Read-only file system..." error:
`singularity exec sandbox/intermediate /usr/msfind/appIwant/binary program [options]`
runs fine:
```
sudo singularity shell --writable sandbox/intermediate
/usr/msfind/appIwant/binary program [options including /run/data test data inside sandbox]
exit
singularity exec sandbox/intermediate /usr/msfind/appIwant/binary program [options]
```

So, basically, after running the app inside the sandbox, it runs fine with exec and I can build an SIF, etc...
I've tried to chmod the dlls and other files in the appIwant folder to give them the permissions they have in the final working sandbox/SIF built from that sandbox and I still get the Read-only file system error. I've tried running the app on my test data in the %test section of the recipe but I get exit 255 failure during build. Anything else I can try? My singularity version is 3.1.0

**Comment by dtrudg** [comment]
Hello, as this is an issue with building a container for a specific application it'd be better addressed in the community Slack channel or mailing list. 

> Note that usage questions, or problems related to running a specific piece of software in a container, are best asked on the Google Group, or Slack channel. Questions in these venues will be seen by a greater number of users, who may already know the answer!
https://github.com/sylabs/singularity/blob/master/SUPPORT.md

As a quick answer here - the first time you run the program it is trying to create some files, but a SIF image is immutable - nothing new can be written to it after the build. You may want to consider running the program once in the build, so it can create the files and they are present in the SIF. Otherwise look at the `--writable-tmpfs` option. This will allow files to be written into a temporary space that is discarded when the container exits.

Please also note that Singularity 3.1.0 is not supported, and is subject to various security issues fixed in newer versions. I would strongly recommend upgrading.

**Comment by Phylloxera** [comment]
sorry :( I had forgot where I got support before and couldn't think what CE meant. Having our sysadmin update our servers to 3.74+ and updating myself for builds. thanks much @dtrudg !
